services/ui-src/src/utils/autosave/autosave.ts

```diff
@@ -37,45 +35,47 @@ export const autosaveFieldData = async ({
   const { id, updateReport } = report;
   const { userName, state } = user;
 
-  // if field value hasn't changed from database value, don't autosave field
-  const onlyChangedFields = fields.filter(
-    // TODO: do we need a deeper equality check here?
-    (field: FieldInfo) => {
-      const { value, hydrationValue, defaultValue } = field;
-      return value !== defaultValue && !isEqual(value, hydrationValue);
-    }
+  // for each passed field, format for autosave payload (if changed)
+  const fieldsToSave: FieldDataTuple[] = await Promise.all(
+    fields
+      .filter((field: FieldInfo) => {
+        /*
+         * If the field doesn't have a default value, compare against the hydration value
+         * (DynamicFields has this case)
+         */
+        if (!field?.defaultValue) return field.value !== field.hydrationValue;
+        return (
+          // Handles most cases where a user wants to update a field
+          field.value !== field.defaultValue ||
+          // Handles case where a user deletes their entry and blurs out of the field
+          (field.value === field.defaultValue &&
+            field.hydrationValue !== undefined &&
+            field.value !== field.hydrationValue)
+        );
```

Comment:
Reverted your code back to this cleanup and then expanded on it with an if statement that should handle any case a user is trying to update a field and we should be notified about it.